zulipterminal/ui_tools/buttons.py

```diff
@@ -359,6 +365,22 @@ def _decode_stream_data(encoded_stream_data: str) -> DecodedStream:
             stream_name = hash_util_decode(encoded_stream_data)
             return DecodedStream(stream_id=None, stream_name=stream_name)
 
+    @staticmethod
+    def _decode_pm_data(encoded_pm_data: str) -> DecodedPM:
+        """
+        Returns a dict with PM type and IDs of PM recipients.
+        """
+        # Recipient ids (seperated by `,`) are of prime interest.
+        recipient_ids, *_ = encoded_pm_data.split('-')
+        recipient_ids_list = list(map(int, recipient_ids.split(',')))
+
+        # Currently webapp uses `pm` and `group` suffix interchangeably.
+        # Treat more-than-2 pms to group pms to avoid confusion.
+        pm_type = 'pm' if len(recipient_ids_list) < 3 else 'group'
```

Comment:
Are we defining `pm` and `group` properly here?
The URL doesn't include the *current-user's* `user_id`. So, if the length of `recipient_ids_list` is `1` the number of persons involved would be `2`(i.e. *current-user* and another user). This should be called `pm`, right?
Similarly, if the length of `recipient_ids_list` is `>= 2`, then number of persons involved would be `1 + len(recipient_ids_list)`, and this should be called `group`?

Correct me if I am wrong, please. :)

Reply:
You're right! To clarify, the currently exposed formats of url _may or may not_ contain the current user's _user_id_ (manually tested this locally).

I'll add a conditional check to confirm if recipient_ids contain user id, and then calculate length accordingly. That should clear that edge case.

Thanks for poiting that out :)

Reply:
I think the URL will contain the current user's `user_id` only if the user is conversing with himself?
Or, were you able to discover some other edge-case for this while testing?

Reply:
The Zuilp server has exposed many URL narrow formats as of now (maybe a potential caveat :) ). Try narrowing to a PM by removing your id from the list, you'd still end up narrowing. I discovered that and some more easter eggs during manual testing ;)